### PR TITLE
Document caching impersonation configuration options

### PR DESCRIPTION
**Pull request by findinpath**
## Description

https://github.com/trinodb/trino/blob/b519bdf7d90aeb414121788999b48930cef6d934/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/ImpersonationCachingConfig.java#L34-L39

https://github.com/trinodb/trino/blob/b519bdf7d90aeb414121788999b48930cef6d934/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/ImpersonationCachingConfig.java#L47-L52

### CODE REVIEW

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 1]
```diff
@@ -80,6 +80,14 @@ are also available. They are discussed later in this topic.
...
   - Maximum threads used to refresh cached metastore data.
   - `10`
+* - `hive.user-metastore-cache-ttl`
+  - Duration of how long cached metastore statistics which are user specific
+    in user impersonation scenarios are considered valid.
+  - `10`
+* - `hive.user-metastore-cache-maximum-size`
+  - Maximum number of metastore data objects in the Hive metastore cache
+    which are user specific in user impersonation scenarios .
+  - `10`
```
Comment (ebyhr) [inline docs/src/main/sphinx/object-storage/metastores.md:90]: Could you change `10` to `1000`? 
https://github.com/trinodb/trino/blob/b519bdf7d90aeb414121788999b48930cef6d934/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/ImpersonationCachingConfig.java#L26

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 2]
```diff
@@ -80,6 +80,14 @@ are also available. They are discussed later in this topic.
 * - `hive.metastore-refresh-max-threads`
   - Maximum threads used to refresh cached metastore data.
   - `10`
+* - `hive.user-metastore-cache-ttl`
+  - Duration of how long cached metastore statistics which are user specific
+    in user impersonation scenarios are considered valid.
+  - `10`
```
Comment (ebyhr) [inline docs/src/main/sphinx/object-storage/metastores.md:86]: Could you change `10` to `0s`? 
https://github.com/trinodb/trino/blob/b519bdf7d90aeb414121788999b48930cef6d934/plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/cache/ImpersonationCachingConfig.java#L25

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 3]
```diff
@@ -80,6 +80,14 @@ are also available. They are discussed later in this topic.
...
+* - `hive.user-metastore-cache-maximum-size`
+  - Maximum number of metastore data objects in the Hive metastore cache
+    which are user specific in user impersonation scenarios .
```
Comment (ebyhr) [inline docs/src/main/sphinx/object-storage/metastores.md:89]:
```suggestion
    which are user specific in user impersonation scenarios.
```

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 4]
```diff
@@ -80,6 +80,14 @@ are also available. They are discussed later in this topic.
 * - `hive.metastore-refresh-max-threads`
   - Maximum threads used to refresh cached metastore data.
   - `10`
+* - `hive.user-metastore-cache-ttl`
+  - Duration of how long cached metastore statistics which are user specific
```
Comment (ebyhr) [inline docs/src/main/sphinx/object-storage/metastores.md:84]: Could you change `Duration` to `[Duration](prop-type-duration)`?

**Review by ebyhr — APPROVED** [review]

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 5]
```diff
@@ -80,6 +80,14 @@ are also available. They are discussed later in this topic.
 * - `hive.metastore-refresh-max-threads`
   - Maximum threads used to refresh cached metastore data.
   - `10`
+* - `hive.user-metastore-cache-ttl`
+  - [Duration](prop-type-duration) of how long cached metastore statistics which are user specific
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/metastores.md:84]:
```suggestion
  - [Duration](prop-type-duration) of how long cached metastore statistics, which are user specific
```

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 6]
```diff
@@ -80,6 +80,14 @@ are also available. They are discussed later in this topic.
...
+    in user impersonation scenarios are considered valid.
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/metastores.md:85]:
```suggestion
    in user impersonation scenarios, are considered valid.
```

File: docs/src/main/sphinx/object-storage/metastores.md [hunk 7]
```diff
@@ -80,6 +80,14 @@ are also available. They are discussed later in this topic.
...
+  - `10s`
+* - `hive.user-metastore-cache-maximum-size`
+  - Maximum number of metastore data objects in the Hive metastore cache
```
Comment (mosabua) [inline docs/src/main/sphinx/object-storage/metastores.md:88]:
```suggestion
  - Maximum number of metastore data objects in the Hive metastore cache,
```

**Review by mosabua — CHANGES_REQUESTED** [review]

**Comment by findinpath** [comment]
@mosabua  pls review again this PR

**Review by mosabua — APPROVED** [review]

**Pull request merged**